### DISCUSSION TITLE
Is Bitwuzla a DPLL(T) Based SMT Solver?

### DISCUSSION THREAD

**Discussion by ShuangyuLyu** [bitwuzla/bitwuzla#147] [General] [answered]
Hi,

I am currently working with Bitwuzla and I have a question regarding its architecture. Is Bitwuzla based on the DPLL(T) approach for SMT solving? I would appreciate any insights into its underlying solving strategy and whether it leverages DPLL(T) or uses a different approach.

Thanks in advance!

**Comment by aniemetz** [comment, marked as answer]
Bitwuzla is not a DPLL(T)-based SMT solver, it's architecture is based on a counter example guided abstraction refinement (CEGAR) based technique called lemmas on demand. A more in-depth description of its architecture and features was published at CAV in 2023 and is available here:  https://link.springer.com/chapter/10.1007/978-3-031-37703-7_1.
